cli/command/telemetry_utils.go

```diff
@@ -15,9 +15,10 @@ import (
 )
 
 // BaseMetricAttributes returns an attribute.Set containing attributes to attach to metrics/traces
-func BaseMetricAttributes(cmd *cobra.Command) attribute.Set {
+func BaseMetricAttributes(cmd *cobra.Command, cli *DockerCli) attribute.Set {
 	attrList := []attribute.KeyValue{
 		attribute.String("command.name", getCommandName(cmd)),
+		attribute.Bool("terminal", cli.Out().IsTerminal()),
```

Comment:
I was unsure whether to add this here or to the default resource options, as iiuc we treat one "CLI invocation" as a unique resource (and I guess whether that invocation was interactive/from a terminal or not is an intrinsic attribute of that resource), so open to suggestions

Reply:
The key `terminal` _might_ be confusing here. I'm not sure in which cases `IsTerminal()` would be false so I don't have a strong opinion here, but it was confusing to me as the docker cli would mostly be executed in a shell even if it isn't an interactive session (script).

Reply:
Happy to be corrected, but generally "terminal" in this context is used interchangeably with "interactive terminal". Even when invoked from a script, what we're talking about (and what `IsTerminal` checks for) is whether the file descriptors attached to the process represent a tty or not.

Other than that,

> docker cli would mostly be executed in a shell even if it isn't an interactive session (script)

I'm not super sure this is accurate, although I guess we can't reliably tell. There's a lot of processes that directly invoke the CLI binary though.

Reply:
If you are specifically looking for an interactive terminal, then stdin/stdout would both need to be a terminal for that, but I guess it really depends on what you are looking for here.
Maybe list all the stdios?

Reply:
> Maybe list all the stdios?

In an attribute, independently? That sounds like a good (unopinionated) compromise to me, then downstream processors can decide how to process that information/what constitutes an "interactive" execution.

Reply:
I think I agree with @cpuguy83; an `isatty()` check on each FD might be the most informative (and any tools parsing this data can make decisions based on their own rules, not subject to our opinions in the CLI)...

Reply:
An attribute is likely better here than a resource attribute if the attribute is specific to the one metric.

For the naming, I'd suggest trying to namespace it into something like `command.input.type=tty` or `command.input.type=noninteractive`. I prefer more explicit names to booleans because they're easier to expand and more self-documenting.

For the naming, the way I came up with that is the attribute is related to how the command is invoked. It's related to what type of input is being used. Then it's a general category for the input. This way, if we later decide this format is wrong, we aren't taking a top-level namespace like "terminal" or "command.input" or if we want to add more detail to it, we can do "command.input.user.name" if we wanted the user who invoked the command.

"input" may not be the best name though.

Reply:
I'd suggest going with something like `command.stdin.isatty=true/false` (and also having matching keys for `stdout` and `stderr`) myself; no idea as to the differences between attributes and resouce attributes, but what @jsternberg says sounds reasonable.

Reply:
@jsternberg what do you think about collecting information about the stdios individually (such as `command.stdio.in`, `command.stdio.out`, `command.stdio.err`)?

> An attribute is likely better here than a resource attribute if the attribute is specific to the one metric.

This depends on how we frame it, right? Either the stdios/interactive env is a property of the resource (the CLI invocation, not related to the "command.duration" or any other metric) or it is a property of that specific metric (is tty unique to this/any metric?)

Reply:
I think names like `command.stdin`, `command.stdout`, and `command.stderr` are fine. `command.stdin.isatty` seems fine to me too if we want to go with a boolean.

Reply:
I think `isatty` makes it unambiguous what is being reported and prevents overloading it with a bad enum in the future.

Reply:
+1 for the `isatty` approach for the individual stdios, as individual attributes. Generally i'd try to process as little as possible and keep the data raw and unopinionated. Any further processing can be done collector-side.